### ISSUE TITLE
Handle API differences between DataFrame and Series

### ISSUE THREAD

**Issue by mrocklin** [dask/dask-expr#98]
Right now we implement type-agnostic methods on `Expr` and then have `DataFrame` and `Series` mostly defer to `Expr` when they don't have something.  This has been effective at getting us pretty far with low boiler-plate and also while being decently (but not entirely) simple.  

However sometimes this doesn't work, as with operations like `value_counts` or `unique` or `drop_duplicates` where DataFrame and Series have different APIs or in operations like `nlargest` where keywords differ based on the DataFrame/Series difference.

What should we do?

**Comment by mrocklin** [comment]
One option that comes to mind for operations like `value_counts` that are on one class but not the other is that we could register the class somewhere:

```python
@Series.register
class ValueCounts(Expr):
    ...
```

We would then auto-generate a method, or dynamically select the class from within some `getattr` call:

```python
class Series:

    def register(self, operation):
        self.methods[normalize(operation.__name__)] = operation

    def __getattr__(self, key):
        if key in Series.methods:
            return functools.partial(Series.methods[key], self)
        ...
```

This would reduce code duplication (we wouldn't even need to implement a `value_counts` method) but feels maybe a little too magical to me.  

It would also allow us to select which operations were accessible from which classes.

**Comment by mrocklin** [comment]
Another maybe simpler option is that we just stop putting these methods on the `Expr` class, and instead rely more on the `Series/DataFrame` classes for API.

```python
class ValueCounts(Expr):
    ...

class Series:
    def value_counts(...):
        ...
```

We can no longer call `expr.value_counts(...)` but we weren't really doing that anyway.

**Comment by phofl** [comment]
I think I prefer putting them on ``Series/DataFrame`` for now, we can always revisit if it ends up getting into a state where it's messy?

**Comment by mrocklin** [comment]
@rjzamora are you ok with this decision?

**Comment by rjzamora** [comment]
>We can no longer call expr.value_counts(...) but we weren't really doing that anyway.

I would agree with the "we weren't really doing that anyway", and go a bit further and say "we should avoid doing that anyway." It feels clearest to me that we focus on classes at the `Expr` level (like `ValueCounts(expr, ...)`), and only use/implement most pandas-like methods at the collection level.
